### PR TITLE
Add configurable schedule name prefix

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
This removes the hard coded schedule name prefix and makes it configurable.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
For multiple deployments of flyteadmin (e.g. a staging, production, etc) the same schedule name prefix is applied without configuration across deployments. This means enabling or disabling a schedule in one deployment has the effect of adding or removing across all deployments such that any matching workflows will be scheduled/unscheduled.

## Tracking Issue
https://github.com/lyft/flyte/issues/227

## Follow-up issue
_NA_

### CODE REVIEW

File: flyteadmin_config.yaml [hunk 1]
```diff
@@ -52,6 +52,7 @@ scheduler:
     region: "my-region"
     scheduleRole: "arn:aws:iam::abc123:role/my-iam-role"
     targetName: "arn:aws:sqs:my-region:abc123:my-queue"
+    scheduleNamePrefix: "flyte"
```
Comment (kumare3) [inline flyteadmin_config.yaml]: can we make this optional? and the default can be `flyte`

File: pkg/runtime/interfaces/application_configuration.go [hunk 2]
```diff
@@ -31,10 +31,11 @@ type ApplicationConfig struct {
 }
 
 type EventSchedulerConfig struct {
```
Comment (kumare3) [inline pkg/runtime/interfaces/application_configuration.go]: Documentation please

File: pkg/runtime/interfaces/application_configuration.go [hunk 3]
```diff
@@ -31,10 +31,11 @@ type ApplicationConfig struct {
...
-	Scheme       string `json:"scheme"`
-	Region       string `json:"region"`
-	ScheduleRole string `json:"scheduleRole"`
-	TargetName   string `json:"targetName"`
+	Scheme             string `json:"scheme"`
```
Comment (kumare3) [inline pkg/runtime/interfaces/application_configuration.go]: Can we document every element in the config?

File: flyteadmin_config.yaml (hunk 1 above)
Comment (katrogan) [reply flyteadmin_config.yaml]: It is optional, if it's empty we just don't join "<prefix>_" with the identifier hash

**Comment by anandswaminathan** [comment]
+1

Internally we have worked around this using different launch plan names.

File: flyteadmin_config.yaml (hunk 1 above)
Comment (kumare3) [reply flyteadmin_config.yaml]: but i think having "flyte_" is better than just an identifier hash. In some organizations, this will help them figure out how the schedules were created

Comment (katrogan) [reply flyteadmin_config.yaml]: the schedule description includes a descriptive identifier that pinpoints it's a workflow execution e.g. `Schedule for Project:flytekit Domain:production Name:workflows.app.MyWorkflow`

**Comment by katrogan** [comment]
@kumare3 added documentation to the config file, PTAL

File: pkg/runtime/interfaces/application_configuration.go (hunk 2 above)
Comment (katrogan) [reply pkg/runtime/interfaces/application_configuration.go]: done

**Comment by kumare3** [comment]
Thank you @katrogan - your rock!
